Index: qremote/qremote.c
===================================================================
--- qremote/qremote.c	(revision 519)
+++ qremote/qremote.c	(working copy)
@@ -431,6 +431,10 @@
 			quitmsg();
 			continue;
 		}
+		while (!strncmp("220-", linein, 4)) {
+			if (netget() != 220)
+				break;
+		}
 		if (linein[3] != ' ') {
 			const char *dropmsg[] = {"invalid greeting from ", NULL, NULL};
 
