Index: qsmtpd/addrsyntax.c
===================================================================
--- qsmtpd/addrsyntax.c	(revision 745)
+++ qsmtpd/addrsyntax.c	(working copy)
@@ -83,7 +83,7 @@
 	if (!at)
 		return 1 - domainvalid(addr);
 	/* localpart too long */
-	if ((at - addr) > 64)
+	if ((at - addr) > 256)
 		return 0;
 	/* paranoid ones would check for (at == addr+i) here */
 	if (parselocalpart(addr) < 0)
