Index: qremote.c
===================================================================
--- qremote.c	(Revision 856)
+++ qremote.c	(Arbeitskopie)
@@ -268,14 +268,14 @@
 	if (status) {
 		int m;
 
-		if ((res >= 211) && (res <= 252)) {
+		if ((res >= 200) && (res <= 299)) {
 			if (status[0] == ' ') {
 				ignore = 1;
 			} else {
 				write(1, status, 1);
 			}
 			m = 1;
-		} else if ((res >= 421) && (res <= 452)) {
+		} else if ((res >= 400) && (res <= 499)) {
 			write(1, status + 1, 1);
 			m = 2;
 		} else {
