diff -aur vpopmail-5.4.0/Makefile.in vpopmail-5.4.0-eike/Makefile.in
--- vpopmail-5.4.0/Makefile.in	2004-02-02 06:24:19.000000000 +0100
+++ vpopmail-5.4.0-eike/Makefile.in	2004-09-26 14:21:29.000000000 +0200
@@ -784,7 +784,7 @@
 	done
 
 install-data-local:
-	$(INSTALL) -d -g @vpopgroup@ -m 0700 -o @vpopuser@ \
+	$(INSTALL) -d -g @vpopgroup@ -m 0755 -o @vpopuser@ \
 	  $(DESTDIR)@vpopmaildir@/@domains_dir@
 
 	echo "-I$(DESTDIR)@vpopmaildir@/include" > \
diff -aur vpopmail-5.4.0/vpopmail.c vpopmail-5.4.0-eike/vpopmail.c
--- vpopmail-5.4.0/vpopmail.c	2004-01-13 16:59:42.000000000 +0100
+++ vpopmail-5.4.0-eike/vpopmail.c	2004-09-26 14:47:46.000000000 +0200
@@ -134,7 +134,7 @@
   if ( chdir(DOMAINS_DIR) != 0 ) {
 
     /* if it's not there, no problem, just try to create it */
-    if ( mkdir(DOMAINS_DIR, VPOPMAIL_DIR_MODE) != 0 ) {
+    if ( mkdir(DOMAINS_DIR, VPOPMAIL_PUBLIC_DIR_MODE) != 0 ) {
       chdir(calling_dir);
       return(VA_CAN_NOT_MAKE_DOMAINS_DIR);
     }
@@ -1728,7 +1728,7 @@
   }
 
   /* create the users dir, including all the Maildir structure */ 
-  if ( mkdir(username, VPOPMAIL_DIR_MODE) != 0 ) {
+  if ( mkdir(username, VPOPMAIL_PUBLIC_DIR_MODE) != 0 ) {
     /* need to add some code to remove the hashed dirs we created above... */
     verrori = VA_EXIST_U_DIR;
     chdir(calling_dir);
@@ -1791,7 +1791,7 @@
   for(i=0;path[i]!=0;++i){
     if ( (i > 0) && (path[i] == '/') ) {
       tmpbuf[i] = 0;
-      if (mkdir(tmpbuf,VPOPMAIL_DIR_MODE) == 0)
+      if (mkdir(tmpbuf,VPOPMAIL_PUBLIC_DIR_MODE) == 0)
         chown(tmpbuf, uid, gid);
     }
     tmpbuf[i] = path[i];
diff -aur vpopmail-5.4.0/vpopmail.h vpopmail-5.4.0-eike/vpopmail.h
--- vpopmail-5.4.0/vpopmail.h	2003-12-17 05:21:10.000000000 +0100
+++ vpopmail-5.4.0-eike/vpopmail.h	2004-09-26 14:47:15.000000000 +0200
@@ -39,6 +39,7 @@
 #define VPOPMAIL_UMASK          0077
 #define VPOPMAIL_TCPRULES_UMASK 0022
 #define VPOPMAIL_DIR_MODE       0750
+#define VPOPMAIL_PUBLIC_DIR_MODE       0755
 #define VPOPMAIL_QMAIL_MODE S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH
 
 #define USE_POP      0x00
