diff -Naurp vpopmail-5.4.30-orig//Makefile.in vpopmail-5.4.30/Makefile.in
--- vpopmail-5.4.30-orig//Makefile.in	2009-12-10 20:33:49.000000000 +0100
+++ vpopmail-5.4.30/Makefile.in	2010-09-08 21:59:41.000000000 +0200
@@ -1058,7 +1058,7 @@ install-exec-am:
 	done
 
 install-data-local:
-	$(INSTALL) -d -g @vpopgroup@ -m 0700 -o @vpopuser@ \
+	$(INSTALL) -d -g @vpopgroup@ -m 0755 -o @vpopuser@ \
 	  $(DESTDIR)@vpopmaildir@/@domains_dir@
 
 	echo "-I@vpopmaildir@/include" > \
diff -Naurp vpopmail-5.4.30-orig//vpopmail.c vpopmail-5.4.30/vpopmail.c
--- vpopmail-5.4.30-orig//vpopmail.c	2009-12-10 20:33:49.000000000 +0100
+++ vpopmail-5.4.30/vpopmail.c	2010-09-08 22:01:51.000000000 +0200
@@ -163,7 +163,7 @@ int vadddomain( char *domain, char *dir,
   if ( chdir(DOMAINS_DIR) != 0 ) {
 
     /* if it's not there, no problem, just try to create it */
-    if ( mkdir(DOMAINS_DIR, VPOPMAIL_DIR_MODE) != 0 ) {
+    if ( mkdir(DOMAINS_DIR, VPOPMAIL_PUBLIC_DIR_MODE) != 0 ) {
       chdir(calling_dir);
       return(VA_CAN_NOT_MAKE_DOMAINS_DIR);
     }
@@ -2385,7 +2385,7 @@ char *make_user_dir(char *username, char
   }
 
   /* create the users dir, including all the Maildir structure */ 
-  if ( mkdir(username, VPOPMAIL_DIR_MODE) != 0 ) {
+  if ( mkdir(username, VPOPMAIL_PUBLIC_DIR_MODE) != 0 ) {
     /* need to add some code to remove the hashed dirs we created above... */
     verrori = VA_EXIST_U_DIR;
     chdir(calling_dir);
@@ -2452,7 +2452,7 @@ int r_mkdir(char *path, uid_t uid, gid_t
   for(i=0; ;++i){
     if ( (i > 0) && ((path[i] == '/') || (path[i] == '\0')) ) {
       tmpbuf[i] = 0;
-      err = mkdir(tmpbuf,VPOPMAIL_DIR_MODE);
+      err = mkdir(tmpbuf,VPOPMAIL_PUBLIC_DIR_MODE);
       if (err == 0)
         chown(tmpbuf, uid, gid);
       else if (errno != EEXIST) {
diff -Naurp vpopmail-5.4.30-orig//vpopmail.h vpopmail-5.4.30/vpopmail.h
--- vpopmail-5.4.30-orig//vpopmail.h	2009-12-10 20:33:49.000000000 +0100
+++ vpopmail-5.4.30/vpopmail.h	2010-09-08 21:59:41.000000000 +0200
@@ -54,6 +54,7 @@
 #define VPOPMAIL_UMASK          0077
 #define VPOPMAIL_TCPRULES_UMASK 0022
 #define VPOPMAIL_DIR_MODE       0750
+#define VPOPMAIL_PUBLIC_DIR_MODE       0755
 #define VPOPMAIL_QMAIL_MODE S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH
 
 #define USE_POP      0x00
